misk/src/main/kotlin/misk/web/exceptions/ExceptionMappingInterceptor.kt

```diff
@@ -0,0 +1,57 @@
+package misk.web.exceptions
+
+import com.google.common.util.concurrent.UncheckedExecutionException
+import misk.Action
+import misk.Chain
+import misk.Interceptor
+import misk.exceptions.StatusCode
+import misk.web.Response
+import misk.web.marshal.StringResponseBody
+import misk.web.mediatype.MediaTypes
+import okhttp3.Headers
+import java.lang.reflect.InvocationTargetException
+import java.util.concurrent.ExecutionException
+import javax.inject.Inject
+
+/**
+ * Converts application and component level dispatch exception into the appropriate response
+ * format. Allows application and component code to control how exceptions are represented to
+ * clients; for example by setting the status code appropriately, or by returning a specialized
+ * response format specific to the error. Components can control how exceptions are mapped
+ * by installing [ExceptionMapper] via the [ExceptionMapperModule]
+ */
+class ExceptionMappingInterceptor(private val mappers: Set<ExceptionMapper<*>>) : Interceptor {
```

Comment:
Should exception mapping and logging be a single interceptor? Thinking being we can avoid accidentally swallowing or duplicating exceptions

Reply:
Yeah, debated this while writing the code. Probably a good idea.